### PR TITLE
Allow ScrollViewer to scroll the content up, down, left and right by …

### PR DESCRIPTION
**Pull request by workgroupengineering**
Allow ScrollViewer to scroll the content up, down, left and right by one page.

## What does the pull request do?

Add PageUp, PageDown, PageLeft and PageRight methods.

## What is the current behavior?

not support scroll by one page

## Checklist

- [ ] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by kekekeks** [comment]
It seems that in WPF those methods live in [ScrollContentPresenter](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.scrollcontentpresenter?view=netcore-3.1).

**Comment by workgroupengineering** [comment]
> 
> 
> It seems that in WPF those methods live in [ScrollContentPresenter](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.scrollcontentpresenter?view=netcore-3.1).

Yes, but the architecture is completely different. The management is entrusted to each single panel by implementing the [IScrollInfo](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.primitives.iscrollinfo?view=netcore-3.1) interface. I have adapted to the existing architecture of Avalonia.

**Comment by kekekeks** [comment]
@grokys

File: src/Avalonia.Controls/ScrollViewer.cs [hunk 1]
```diff
@@ -448,6 +448,38 @@ public void LineRight()
             Offset += new Vector(_smallChange.Width, 0);
         }
 
+        /// <summary>
+        /// Scrolls the content upward by one page.
+        /// </summary>
+        public void PageUp()
```
Comment (kekekeks) [inline src/Avalonia.Controls/ScrollViewer.cs:454]: This code doesn't seem to support ILogicalScrollable. See ScrollContentPresenter.cs sources for more details.

**Comment by grokys** [comment]
> It seems that in WPF those methods live in ScrollContentPresenter.

They're also present in WPF's ScrollViewer: https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.scrollviewer.pageup?view=netcore-3.1

> why do we have LineUp/Down/Left/Right in ScrollViewer while everything else lives in ScrollContentPresenter?

What do you mean by "everything else" here? SCP doesn't contain any utility methods like this in Avalonia. This is because SCP is usually hidden away in a template and the user-visible API is `ScrollViewer`.

File: src/Avalonia.Controls/ScrollViewer.cs (hunk 1 above)
Comment (grokys) [reply src/Avalonia.Controls/ScrollViewer.cs:454]: I think this should handle `ILogicalScrollable`. The logical scrolling stuff should be handled in `ScrollContentPresenter` and by the time the info gets to `ScrollViewer` (via two-way bindings) the offset/extent/viewport will have been calculated according to logical scrolling.

Comment (workgroupengineering) [reply src/Avalonia.Controls/ScrollViewer.cs:454]: @grokys Practically. Do you suggest replicating the methods in the ILogicalScrollable interface and using these in the ScrollViewer?

Comment (grokys) [reply src/Avalonia.Controls/ScrollViewer.cs:454]: Not 100% sure what you mean, but I don't think I was suggesting that, no ;)

**Review by grokys — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**